a10_octavia/controller/worker/tasks/decorators.py

```diff
@@ -14,14 +14,33 @@
 
 
 import acos_client
+from acos_client.errors import ACOSException
+from acos_client.errors import Exists
+
+try:
+    from http.client import BadStatusLine
+except ImportError:
+    from httplib import BadStatusLine
 from oslo_config import cfg
 from oslo_log import log as logging
 from requests.exceptions import ConnectionError
+from requests.exceptions import ReadTimeout
 
 CONF = cfg.CONF
 LOG = logging.getLogger(__name__)
 
 
+def configure_partition(vthunder_client, partition):
+    try:
+        try:
+            vthunder_client.system.partition.create(partition)
```

Comment:
Making this call every time is a bit much. The only situation in which this should be called is when the load balancer is being created. Let's shift this to a vthunder task called "HandleACOSPartitionChange" or something similar.

This way too we can revert the creation of the partition should an error occur during the flow.

Reply:
Addresses in https://github.com/a10networks/a10-octavia/pull/70